src/container.ts

```diff
@@ -0,0 +1,51 @@
+// To organize imports in a better way
+/* eslint-disable import/order */
+import { Container } from 'inversify';
+
+import ApiService from '#src/services/api.service';
+import WatchHistoryService from '#src/services/watchhistory.service';
+import EpgService from '#src/services/epg.service';
+import EntitlementService from '#src/services/entitlement.service';
+import FavoritesService from '#src/services/favorites.service';
+import ConfigService from '#src/services/config.service';
+
+
+import ApiController from '#src/stores/ApiController';
+import WatchHistoryController from '#src/stores/WatchHistoryController';
+import CheckoutController from '#src/stores/CheckoutController';
+import AccountController from '#src/stores/AccountController';
+import EpgController from '#src/stores/EpgController';
+import EntitlementController from '#src/stores/EntitlementController';
+import ProfileController from '#src/stores/ProfileController';
+import FavoritesController from '#src/stores/FavoritesController';
+import AppController from '#src/stores/AppController';
+import { SettingsController } from '#src/stores/SettingsController';
+import type { interfaces } from 'inversify';
+
+export const container = new Container({ defaultScope: 'Singleton', skipBaseClassChecks: true });
+
+// resolve shortcut
+export const getModule = <T>(constructorFunction: interfaces.Newable<T>): T => {
+  return container.resolve(constructorFunction);
```

Comment:
You might be right that we don't need all of our custom code for the resolution. @AntonLantukh can we try simplifying it and see if you branch still works?

Reply:
Hm, we might still want to avoid calling and creating controllers which are not supposed to exist / work (like CheckoutController for StartWatchingButton with AVOD).

Reply:
Ah, good point!

Reply:
But that is different from what the code does, right? It checks the uniqueness of a service but does not conditionally return a service.

I would expect something like this then:

```ts
export const getModule = <T>(type: interfaces.Newable<T>): T => {
  const all = container.getAll(type);

  // don't resolve when there are no services registered
  if (!all.length) {
    return undefined;
  }

  return container.resolve(type);
};
```

The issue with the above is that all `getModule` calls will return `T | undefined`, and we will have to optional chain every service call.

IMO, we should prevent rendering components/containers that depend on account services conditionally, but I haven't thought that one through well.

Reply:
Maybe the easiest way to do that is to allow getModule to return optionally undefined, and in the components that shouldn't render, simply short-circuit with 
```
if (!module) { return null; }
```

Reply:
What about having a separate `getOptionalModule` to clarify that a service/controller is optional? For most services, we expect it to be present.

Reply:
Danny suggested to use optional `required` property and to throw the error in case it is required and not found:

https://github.com/jwplayer/ott-web-app/pull/363#discussion_r1386573736

Reply:
Yes, I like that solution as well! 😄